Core/src/org/sleuthkit/autopsy/centralrepository/datamodel/CRAccount.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Central Repository
+ *
+ * Copyright 2020 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.centralrepository.datamodel;
+
+import org.sleuthkit.datamodel.Account;
+
+
+/**
+ * This class abstracts an Account as stored in the CR database.
+ */
+public final class CRAccount {
+    
+	// primary key in the Accounts table in CR database
+	private final long account_id;
+
+	private final CRAccountType crAccountType;
+	private final String typeSpecificID;
+        
+    /**
+     * A CRAccounType encapsulates an account type and the correlation type
+     * that it maps to.
+     */
+    public static final class CRAccountType {
+
+        // id is the primary key in the account_types table
+        private final int crAccountTypeID;
+        private final Account.Type acctType;
+        private final int correlationTypeID;
+
+        CRAccountType(int acctTypeID, Account.Type acctType, int correlation_type_id) {
+            this.acctType = acctType;
+            this.correlationTypeID = correlation_type_id;
+            this.crAccountTypeID = acctTypeID;
+        }
+        
+
+        /**
+         * @return the acctType
+         */
+        public Account.Type getAcctType() {
+            return acctType;
+        }
+
+        public int getCorrelationTypeId() {
```

Comment:
If this is used to just follow up with a query to get the Type instance with this id, can't the CentralRepoAccountType just be initialized with that instance?

Reply:
Yes, currently the only call to getCorrelationTypeId() is followed by getting the type instance.  

But I think as we start using t and build out the UI, the CentralRepoAccountType may be instantiated without needing  the corresponding Correlationtype all the time.  That's why I'm keeping the is so that the CorrelationType can be read in as needed.

Reply:
Ok.